datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxDrService.java

```diff
@@ -0,0 +1,273 @@
+
+    }
+
+    private String triggerDatalakeDatabaseBackupFlow(Long clusterId, String backupId, String backupLocation) {
+        return sdxReactorFlowManager.triggerDatalakeDatabaseBackupFlow(clusterId, backupId, backupLocation);
+    }
+
+    private String triggerDatalakeDatabaseRestoreFlow(Long clusterId, String backupId, String backupLocation) {
+        return sdxReactorFlowManager.triggerDatalakeDatabaseRestoreFlow(clusterId, backupId, backupLocation);
+    }
+
+    public void databaseBackup(SdxDatabaseDrStatus drStatus, Long clusterId, String backupId, String backupLocation) {
+        try {
+            sdxDatabaseDrStatusRepository.save(drStatus);
+            sdxClusterRepository.findById(clusterId).ifPresentOrElse(sdxCluster -> {
+                BackupV4Response backupV4Response = stackV4Endpoint.backupDatabaseByName(0L, sdxCluster.getClusterName(),
+                        backupLocation, backupId);
+                cloudbreakFlowService.saveLastCloudbreakFlowChainId(sdxCluster, backupV4Response.getFlowIdentifier());
+                updateDatabaseStatusEntry(drStatus.getOperationId(), SdxDatabaseDrStatus.Status.TRIGGERRED, null);
+            }, () -> {
+                updateDatabaseStatusEntry(drStatus.getOperationId(), SdxDatabaseDrStatus.Status.FAILED,
+                        String.format("SDX cluster with Id [%d] not found", clusterId));
+                throw notFound("SDX cluster", clusterId).get();
+            });
+        } catch (WebApplicationException e) {
+            String message = String.format("Database backup failed for datalake-id: [%d]. Message: [%s]", clusterId, e.getMessage());
+            throw new CloudbreakApiException(message);
+        }
+    }
+
+    public void databaseRestore(SdxDatabaseDrStatus drStatus, Long clusterId, String backupId, String backupLocation) {
+        try {
+            sdxDatabaseDrStatusRepository.save(drStatus);
+            sdxClusterRepository.findById(clusterId).ifPresentOrElse(sdxCluster -> {
+                RestoreV4Response restoreV4Response = stackV4Endpoint.restoreDatabaseByName(0L, sdxCluster.getClusterName(),
+                        backupLocation, backupId);
+                cloudbreakFlowService.saveLastCloudbreakFlowChainId(sdxCluster, restoreV4Response.getFlowIdentifier());
+                updateDatabaseStatusEntry(drStatus.getOperationId(), SdxDatabaseDrStatus.Status.TRIGGERRED, null);
+            }, () -> {
+                updateDatabaseStatusEntry(drStatus.getOperationId(), SdxDatabaseDrStatus.Status.FAILED,
+                        String.format("SDX cluster with Id %d not found", clusterId));
+                throw notFound("SDX cluster", clusterId).get();
+            });
+        } catch (WebApplicationException e) {
+            String message = String.format("Database restore failed for datalake-id: [%d]. Message: [%s]", clusterId, e.getMessage());
+            throw new CloudbreakApiException(message);
+        }
```

Comment:
you might want to use `WebApplicationExceptionMessageExtractor` here

Reply:
Will use it